crates/goose/src/token_counter.rs

```diff
@@ -68,16 +71,137 @@ impl TokenCounter {
         }
     }
 
-    pub fn count_tokens(&self, text: &str, model_name: Option<&str>) -> usize {
+    fn get_tokenizer(&self, model_name: Option<&str>) -> &Tokenizer {
         let tokenizer_key = Self::model_to_tokenizer_key(model_name);
-        dbg!(&model_name, &tokenizer_key);
-        let tokenizer = self
-            .tokenizers
+        self.tokenizers
             .get(tokenizer_key)
-            .expect("Tokenizer not found");
+            .expect("Tokenizer not found")
+    }
+
+    pub fn count_tokens(&self, text: &str, model_name: Option<&str>) -> usize {
```

Comment:
why don't we make use of the model providers' responses which already have input/output token counts instead of tokenizing all the messages ourselves?

Reply:
we did this in goose V0 and it's hard to get right and keep track of. def an approach to think through more though, if we want to be very accurate. the trade off here accuracy vs. ease of implementation and overhead.